instrumentation/aws-lambda/aws-lambda-core-1.0/library/src/main/java/io/opentelemetry/instrumentation/awslambdacore/v1_0/internal/AwsXRayEnvSpanLinksExtractor.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.instrumentation.awslambdacore.v1_0.internal;
+
+import io.opentelemetry.api.trace.Span;
+import io.opentelemetry.api.trace.SpanContext;
+import io.opentelemetry.context.Context;
+import io.opentelemetry.context.propagation.TextMapGetter;
+import io.opentelemetry.contrib.awsxray.propagator.AwsXrayPropagator;
+import io.opentelemetry.instrumentation.api.instrumenter.SpanLinksBuilder;
+import io.opentelemetry.instrumentation.api.instrumenter.SpanLinksExtractor;
+import io.opentelemetry.instrumentation.awslambdacore.v1_0.AwsLambdaRequest;
+import java.util.Collections;
+import java.util.Locale;
+import java.util.Map;
+
+/**
+ * This class is internal and is hence not for public use. Its APIs are unstable and can change at
+ * any time.
+ */
+public final class AwsXRayEnvSpanLinksExtractor implements SpanLinksExtractor<AwsLambdaRequest> {
+
+  private static final String AWS_TRACE_HEADER_ENV_KEY = "_X_AMZN_TRACE_ID";
+
+  // lower-case map getter used for extraction
+  private static final String AWS_TRACE_HEADER_PROPAGATOR_KEY = "x-amzn-trace-id";
+
+  @Override
+  public void extract(
+      SpanLinksBuilder spanLinks,
+      io.opentelemetry.context.Context parentContext,
+      AwsLambdaRequest awsLambdaRequest) {
+    extract(spanLinks);
+  }
+
+  public static void extract(SpanLinksBuilder spanLinks) {
+    String parentTraceHeader = System.getenv(AWS_TRACE_HEADER_ENV_KEY);
+    if (parentTraceHeader == null || parentTraceHeader.isEmpty()) {
+      return;
+    }
+    Context parentCtx =
+        AwsXrayPropagator.getInstance()
+            .extract(
+                // see BaseTracer#extract() on why we're using root() here
+                Context.root(),
+                Collections.singletonMap(AWS_TRACE_HEADER_PROPAGATOR_KEY, parentTraceHeader),
+                MapGetter.INSTANCE);
+    SpanContext parent = Span.fromContext(parentCtx).getSpanContext();
```

Comment:
Nit pick. It isn't a `parent`, so could be confusing for future readers to see `parentCtx` and `parent`.

Reply:
better now?

Reply:
Yup :+1: